lib/Sema/TypeCheckPattern.cpp

```diff
@@ -775,10 +775,31 @@ static bool validateParameterType(ParamDecl *decl, DeclContext *DC,
   return hadError;
 }
 
+/// Request nominal layout for any types that could be sources of typemetadata
+/// or conformances.
+void TypeChecker::requestRequiredNominalTypeLayoutForParameters(
+    ParameterList *PL) {
+  for (auto param : *PL) {
+    auto type = param->hasType() ? param->getType()->getCanonicalType()
```

Comment:
I think we'll always have a type as this point; did you add the `hasType()` because something broke?

Reply:
Typically, we wouldn't use `getCanonicalType()` here, because we don't actually need it...

Reply:
The check for hasType was necessary when called from typeCheckParameterList. This no longer seems necessary after the changes you suggested.

Reply:
Great!